### ISSUE TITLE
Titles in definitions are not used for validation

### ISSUE THREAD

**Issue by jam13** [rjsf-team/react-jsonschema-form#3732]
### Prerequisites

- [X] I have searched the existing issues
- [X] I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [X] I have read the [documentation](https://rjsf-team.github.io/react-jsonschema-form/docs)
- [X] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net), [Codesandbox.io](https://codesandbox.io) or preferably a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### What theme are you using?

antd

### Version

5.8.1

### Current Behavior

When using `$ref` to point to a field definition, validation errors do not use the field title. e.g. must have required property 'myFieldName'

### Expected Behavior

Errors should use field title. e.g. must have required property 'My Field Name'

### Steps To Reproduce

https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6eyJmaXJzdE5hbWUiOiIiLCJhZ2UiOm51bGwsImJpbyI6IlJvdW5kaG91c2Uga2lja2luZyBhc3NlcyBzaW5jZSAxOTQwIiwicGFzc3dvcmQiOiJub25lZWQiLCJ0ZWxlcGhvbmUiOiIxLTgwMC1LSUNLQVNTIn0sInNjaGVtYSI6eyJ0eXBlIjoib2JqZWN0IiwiZGVmaW5pdGlvbnMiOnsibXlDdXN0b21GaWVsZCI6eyJ0eXBlIjoiYm9vbGVhbiIsIm9uZU9mIjpbeyJjb25zdCI6dHJ1ZSwidGl0bGUiOiJZZXMifSx7ImNvbnN0IjpmYWxzZSwidGl0bGUiOiJObyJ9XSwidGl0bGUiOiJNeSBDdXN0b20gRmllbGQifSwiYW5vdGhlckN1c3RvbUZpZWxkIjp7InR5cGUiOiJib29sZWFuIiwib25lT2YiOlt7ImNvbnN0Ijp0cnVlLCJ0aXRsZSI6IlllcyJ9LHsiY29uc3QiOmZhbHNlLCJ0aXRsZSI6Ik5vIn1dLCJ0aXRsZSI6IkFub3RoZXIgQ3VzdG9tIEZpZWxkIn19LCJyZXF1aXJlZCI6WyJteUN1c3RvbUZpZWxkIiwiYW5vdGhlckN1c3RvbUZpZWxkIl0sInByb3BlcnRpZXMiOnsibXlDdXN0b21GaWVsZCI6eyIkcmVmIjoiIy9kZWZpbml0aW9ucy9teUN1c3RvbUZpZWxkIn0sImFub3RoZXJDdXN0b21GaWVsZCI6eyJ0aXRsZSI6Ik92ZXJyaWRkZW4gVGl0bGUiLCIkcmVmIjoiIy9kZWZpbml0aW9ucy9hbm90aGVyQ3VzdG9tRmllbGQifX19LCJ1aVNjaGVtYSI6e30sInRoZW1lIjoiYW50ZCIsImxpdmVTZXR0aW5ncyI6eyJzaG93RXJyb3JMaXN0IjoidG9wIiwidmFsaWRhdGUiOmZhbHNlLCJkaXNhYmxlZCI6ZmFsc2UsIm5vSHRtbDVWYWxpZGF0ZSI6dHJ1ZSwicmVhZG9ubHkiOmZhbHNlLCJvbWl0RXh0cmFEYXRhIjpmYWxzZSwibGl2ZU9taXQiOmZhbHNlLCJleHBlcmltZW50YWxfZGVmYXVsdEZvcm1TdGF0ZUJlaGF2aW9yIjp7ImFycmF5TWluSXRlbXMiOiJwb3B1bGF0ZSIsImVtcHR5T2JqZWN0RmllbGRzIjoicG9wdWxhdGVBbGxEZWZhdWx0cyJ9fX0=

### Environment

```markdown
Playground
```

**Comment by heath-freenome** [comment]
@jam13 Good catch! Given how busy us few maintainers are, your best bet is to provide a PR with the fix. If you were to update the `transformRJSFValidationErrors()` function on line 48 in `processRawValidationErrors.ts` file in the `@rjsf/validator-ajv8` to call the `findSchemaDefinition()` function if the `parentSchema` contains a `REF_KEY` and use that to get the schema title, you should be able to fix this. It is possible that the `findSchemaDefinition()` function may have a bug that will cause the override title to be lost. If that is the case, please also fix that function on line 49. And provide additional tests that verify the fix. Thanks!

**Comment by jam13** [comment]
> @jam13 Good catch! Given how busy us few maintainers are, your best bet is to provide a PR with the fix. If you were to update the `transformRJSFValidationErrors()` function on line 48 in `processRawValidationErrors.ts` file in the `@rjsf/validator-ajv8` to call the `findSchemaDefinition()` function if the `parentSchema` contains a `REF_KEY` and use that to get the schema title, you should be able to fix this. It is possible that the `findSchemaDefinition()` function may have a bug that will cause the override title to be lost. If that is the case, please also fix that function on line 49. And provide additional tests that verify the fix. Thanks!

Thanks, I will have a go when I get some free time

**Comment by heath-freenome** [comment]
I believe this was fixed with https://github.com/rjsf-team/react-jsonschema-form/pull/4349
